apps/src/util/testJavabuilderWebsocketConnection.js

```diff
@@ -0,0 +1,51 @@
+import $ from 'jquery';
+
+// helper to allow behind the scenes test of javabuilder connection
+// on load of high traffic pages. particularly interested
+// in collecting data on whether schools often block websockets.
+export default function testJavabuilderWebsocketConnection() {
+  try {
+    logToFirehose('started');
+
+    const socket = new WebSocket(
+      'wss://javabuilderbeta.code.org?Authorization=connectivityTest'
+    );
+
+    socket.onopen = function(e) {
+      socket.send('connectivityTest');
+    };
+
+    socket.onmessage = function(message) {
+      if (message.data === 'success') {
+        logToFirehose('success');
+      } else {
+        logToFirehose('unexpected message response');
```

Comment:
the string here ends up being the event name in firehose -- i think we usually exclude spaces from event names (so this would be `unexpected-message-response`. i'm not sure that it matters, but may make querying easier? i don't feel too strongly about this 😄